### PR TITLE
Pair potential

### PR DESCRIPTION
**Pull request by VanNguyen23**

### CODE REVIEW

File: Pairwise_potential.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    def cutoff_correction(self, cutoff, box_object, num_particles,):
+        '''The function corrects interaction energy from energy cutoff.
+
+    Parameters
+    ----------
+    cutoff : float
+        Lennard-Jones potential cutoff distance
+
+    box_object : np.array
```
Comment (b-butler) [inline Pairwise_potential.py]: This is a box object.

File: Pairwise_potential.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    epsilon: float
+        depth of potential well
+
+    rij2 : float
+        square distance between two particles
+
+    Return
+    ------
+
+    Value of Lennard-Jones potential energy between two particles at considered distance
```
Comment (b-butler) [inline Pairwise_potential.py]: A little long

File: Pairwise_potential.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    num_particles : float
+        total number of particles in the box
+
+    Return
+    ------
+
+    e_correction : float
+        correction energy from truncation
+    '''
+        volume = np.power(box_object.length, 3)
```
Comment (b-butler) [inline Pairwise_potential.py]: `volume = box_object.volume`

**Review by b-butler — CHANGES_REQUESTED** [review]

**Review by b-butler — CHANGES_REQUESTED** [review]
Hey can you include a cutoff parameter in your potentials so that the `__call__()` method returns 0 if `rij > cutoff2`

File: Pairwise_potential.py [hunk 4]
```diff
@@ -74,12 +75,20 @@ def cutoff_correction(self, cutoff, box_object, num_particles,):
     """
 
         volume = box_object.volume
-        sig_by_cutoff3 = np.power(self.sigma/cutoff, 3)
+        sig_by_cutoff3 = np.power(self.sigma/cutoff2, 3)
```
Comment (b-butler) [inline Pairwise_potential.py]: We still need to use `cutoff` here but we only have `self.cutoff2` so we first need to calculate `np.sqrt(self.cutoff2)`.

File: Pairwise_potential.py [hunk 5]
```diff
@@ -27,12 +27,13 @@ class LJ(PairwisePotential):
...
     
-    def __init__(self,sigma=1.0,epsilon=1.0):
+    def __init__(self,sigma=1.0,epsilon=1.0, cutoff=2.6):
 
         self.sigma = sigma
         self.epsilon = epsilon
+        self.cutoff2 = cutoff * cutoff
     
-    def __call__(self, rij2):
+    def lennard_jones(self, rij2):
```
Comment (b-butler) [inline Pairwise_potential.py]: We should call this `potential` and add that to the abstract class required methods.

**Review by b-butler — CHANGES_REQUESTED** [review]
It is definitely coming along. See requested changes.

File: Pairwise_potential.py (hunk 5 above)
Comment (VanNguyen23) [reply Pairwise_potential.py]: I don't quite get what you mean. You mean use potential instead of call?

Comment (b-butler) [reply Pairwise_potential.py]: Call the method potential.

File: Pairwise_potential.py [hunk 6]
```diff
@@ -6,7 +6,8 @@ class PairwisePotential(ABC):
     Parameters
     ----------
 
-    rij2 = square distance between two particles
+    rij2 : float
```
Comment (b-butler) [inline Pairwise_potential.py]: This can be a float or numpy array now.

**Review by b-butler — APPROVED** [review]

**Pull request merged**